Resources/public/js/views/services/ez-navigationhubviewservice.js

```diff
@@ -48,7 +75,144 @@ YUI.add('ez-navigationhubviewservice', function (Y) {
         _getViewParameters: function () {
             return {
                 user: this.get('app').get('user'),
+                createNavigationItems: this.get('createNavigationItems'),
+                deliverNavigationItems: this.get('deliverNavigationItems'),
+                optimizeNavigationItems: this.get('optimizeNavigationItems'),
+                matchedRoute: this._matchedRoute(),
             };
         },
+
+        /**
+         * A matched route object from the request. This object will be used by
+         * the navigation hub view to check which navigation item view is
+         * selected. A matched route object is a clone of the application active
+         * route without the service, serviceInstance and callbacks entries and
+         * with an additionnal `parameters` property holding the route
+         * parameters and their values.
+         *
+         * @method _matchedRoute
+         * @protected
+         * @return {Object}
+         */
+        _matchedRoute: function () {
+            var request = this.get('request'),
+                route = request.route,
+                matchedRoute = Y.merge(route);
+
+            matchedRoute.parameters = request.params;
+            delete matchedRoute.service;
+            delete matchedRoute.serviceInstance;
+            delete matchedRoute.callbacks;
+
+            return matchedRoute;
+        },
+
+        /**
+         * Adds a navigation item for the given zone.
+         * Note: adding a navigation item will only work before the first
+         * initialization of the navigation hub. As a result, in a navigation
+         * hub view service plugin, this method should be called only in or from
+         * the initializer method.
+         *
+         * @method addNavigationItem
+         * @param {Object} item a navigation item, see the description of the
+         * *NavigationItems attributes
+         * @param {String} zone the identifier of the zone
+         */
+        addNavigationItem: function (item, zone) {
+            this.get(zone + 'NavigationItems').push(item);
+        },
+
+        /**
+         * Removes a navigation item for the given zone.
+         * Note: removing a navigation item will only work before the first
+         * initialization of the navigation hub. As a result, in a navigation
+         * hub view service plugin, this method should be called only in or from
+         * the initializer method.
+         *
+         * @method removeNavigationItem
+         * @param {String} identifier the identifier of the navigation item to
+         * remove
+         * @param {String} zone the identifier of the zone
+         */
+        removeNavigationItem: function (identifier, zone) {
```

Comment:
an extension that wants to remove a default navigation item. You'll use that in Studio.